lib/faraday/adapter/net_http.rb

```diff
@@ -89,11 +89,15 @@ def with_net_http_connection(env)
       end
 
       def net_http_connection(env)
-        if proxy = env[:request][:proxy]
-          Net::HTTP::Proxy(proxy[:uri].host, proxy[:uri].port, proxy[:user], proxy[:password])
+        proxy = env[:request][:proxy]
+        addr  = env[:url].host
+        port  = env[:url].port || (env[:url].scheme == 'https' ? 443 : 80)
+        # HTTP.Proxy is obsolete, and Net::HTTP will use proxy from env as default
+        if proxy.nil?
+          Net::HTTP.new(addr, port, nil)
```

Comment:
Hi @iMacTia 

Thanks for your comments.

However, sending `nil` as the 3rd parameter to `Net::HTTP.new(addr, port, nil) if proxy.nil?` is the correct behavior.

Why needs to send `nil` as the 3rd parameter to `Net::HTTP.new if proxy.nil?`?
+ Net::HTTP will still use the proxy from system env since the changes made in ruby/ruby@9416bec
+ Faraday has its own implementation regarding to proxy in its net/http adapter

---
Generally speaking, here are the cases configured for proxy in `Net::HTTP` adapter:
+ no proxy key when `Faraday.new`
  - use __ENV['http_proxy']__
+ proxy: nil when `Faraday.new`
  - no proxy
+ proxy: { url: xxx, port: xxx } when `Faraday.new`
  - use __proxy: { url: xxx, port: xxx }__
---

Here are relevant code:
+ new Faraday
```
lib/faraday.rb

def new(url = nil, options = nil)
   block = block_given? ? Proc.new : nil
   options = options ? default_connection_options.merge(options) : default_connection_options.dup
   Faraday::Connection.new(url, options, &block)
end
```

+ Faraday connection
```
@proxy = nil
proxy(options.fetch(:proxy) {
    uri = ENV['http_proxy']
    if uri && !uri.empty?
       uri = 'http://' + uri if uri !~ /^http/i
       uri
    end
})
```
```
def proxy(arg = nil)
   return @proxy if arg.nil?
   @proxy = ProxyOptions.from(arg)
end
```

So per my investigation and fix, it will not break the existing code based on current Faraday without this fix.